Yv 755 governance implement popup when delegating stake

AbdullayevAbdurashid opened the pull request:
https://emurgo.atlassian.net/browse/YV-755?atlOrigin=eyJpIjoiY2VjZWZjYmRmZjg4NDE5OGIzOTc2YzU0ODhmNWJiYzAiLCJwIjoiaiJ9

To test :
Open stakign center- delegate to drap without participating in governance> pop up should appear

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Prompt users not participating in governance with a modal during stake delegation, enabling combined stake+DRep delegation to Yoroi or stake-only, plus related UI/i18n and P2P provider placement updates.
> 
> - **Staking**:
>   - Show governance-required modal when delegating if `!isGovernanceParticipating`.
>   - Offer actions: delegate to Yoroi DRep (builds combined stake + DRep tx via `createCombinedDelegationTxFromWallet` with `GOVERNANCE_YOROI_DREP_ID_HEX`) or delegate stake only (`createDelegationTxFromWallet`).
>   - Transaction build flow refactored to handle pending pool selection and errors (no funds -> `navigateTo.noFunds`).
> - **UI**:
>   - New `GovernanceRequiredModal` (content, footer) with new `GovernanceIllustration` SVG.
> - **i18n**:
>   - Add strings for governance-required modal and wire into `useStrings` and `en-US.json`.
> - **App Structure / P2P**:
>   - Move `P2PConnectionProviderWrapper` under `WithWalletOpened`; remove from `App.tsx` and disable `P2PConnectionStatusBar` in `AppNavigator`.
> - **Misc**:
>   - Minor DashboardScreen refactor (navigation var ordering); no behavior change.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0fda615733cb13e06ab81107b7911c8ca121c048. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

AbdullayevAbdurashid committed:
feat: add GovernanceRequiredModal component with content and footer for delegation options

AbdullayevAbdurashid committed:
feat(StakingCenter): implement governance modal when delegating stake

stackchain commented on mobile/src/ui/GovernanceRequiredModal/GovernanceRequiredModal.tsx:
```diff
@@ -0,0 +1,78 @@
+  const {atoms: ta} = useTheme()
+
+  return (
+    <View style={[a.flex_1, a.align_center, a.justify_between, a.py_lg]}>
```
Modal.Content

AbdullayevAbdurashid replied:
Done

stackchain commented on mobile/assets/img/governance-banner.png:
AFAIK, we should be using only SVG comp

AbdullayevAbdurashid replied:
Done

AbdullayevAbdurashid committed:
refactor(GovernanceRequiredModal): remove governance banner image and replace with GovernanceIllustration component

AbdullayevAbdurashid committed:
refactor(GovernanceRequiredModal): replace View with Modal.Content for improved layout

banklesss commented:
@AbdullayevAbdurashid can you check cursor suggestions and fix conflicts?

jorbuedo committed:
Merge remote-tracking branch 'origin/mobile-experimental' into YV-755-governance-implement-popup-when-delegating-stake

jorbuedo committed:
fix: implement combined stake and governance delegation in governance modal

- Fix GovernanceRequiredModal to use different transaction types for each option
- 'Delegate to Yoroi DRep' now uses createCombinedDelegationTxFromWallet with both poolId and drepValue
- 'Delegate stake only' uses createDelegationTxFromWallet for stake-only delegation
- Add governance participation check to show modal when user hasn't delegated to DRep
- Use existing createCombinedDelegationTxFromWallet recipe for combined transactions

jorbuedo committed:
fix: add missing governance required modal string definitions

- Add governanceRequiredTitle, governanceRequiredDescription, delegateToYoroiDRep, and delegateStakeOnly to staking messages
- Fixes formatjs/intl error about missing message IDs

jorbuedo committed:
fix
